Rubberduck.Inspections/Concrete/RedundantByRefModifierInspection.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Antlr4.Runtime;
+using Rubberduck.Common;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Resources;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.VBEditor;
+
+namespace Rubberduck.Inspections.Concrete
+{
+    public sealed class RedundantByRefModifierInspection : InspectionBase, IParseTreeInspection
+    {
+        public RedundantByRefModifierInspection(RubberduckParserState state)
+            : base(state, CodeInspectionSeverity.Hint)
+        {
+        }
+
+        public override CodeInspectionType InspectionType => CodeInspectionType.CodeQualityIssues;
+
+        public IInspectionListener Listener { get; } = new RedundantByRefModifierListener();
+
+        public override IEnumerable<IInspectionResult> GetInspectionResults()
+        {
+            var builtInEventHandlerContexts = State.DeclarationFinder.FindEventHandlers().Select(handler => handler.Context);
+            var interfaceImplementationMemberContexts = UserDeclarations.FindInterfaceImplementationMembers().Select(member => member.Context);
```

Comment:
For performance reasons, you should probably materialize these two enumerables into hash sets. Then `Contains` is O(1) instead of O(N).